src/lib/configs/project/provider.ts

```diff
@@ -442,57 +513,41 @@ export async function initNewProviderConfig({
   name,
   ...args
 }: ProviderConfigArgs) {
-  if (name !== undefined) {
-    setProviderConfigName(name);
-  }
-
-  const providerConfig = await getConfigInitFunction(
-    getInitConfigOptions(),
-    getDefault(args),
-  )();
-
-  await ensureSecrets(providerConfig);
-  await ensureConfigToml(providerConfig);
-  return providerConfig;
+  return createProviderConfigWithSecretsAndConfigTomls(name, () => {
+    return getConfigInitFunction(getInitConfigOptions(), getDefault(args))();
+  });
```

Comment:
```suggestion
  return createProviderConfigWithSecretsAndConfigTomls(name, getConfigInitFunction(getInitConfigOptions(), getDefault(args)));
```
Will it work?

Reply:
yeah, it should, didn't notice that aspect after refactoring